src/web/client/common/localStore.ts

```diff
@@ -0,0 +1,132 @@
+/* eslint-disable @typescript-eslint/no-unused-vars */
+/* --------------------------------------------------------------------------------------------
+ * Copyright (c) Microsoft Corporation. All rights reserved.
+ * Licensed under the MIT License. See License.txt in the project root for license information.
+ * ------------------------------------------------------------------------------------------ */
+/* eslint-disable @typescript-eslint/no-explicit-any */
+
+'use strict';
+import * as vscode from "vscode";
+import { getHeader } from "./authenticationProvider";
+import { PORTAL_LANGUAGES, PORTAL_LANGUAGES_URL_KEY, PORTAL_LANGUAGE_DEFAULT, WEBPAGEID_URL_KEY, WEBPAGES, WEBSITE_LANGUAGES, WEBSITE_LANGUAGES_URL_KEY } from "./constants";
+import { readSchema } from "./portalSchemaReader";
+
+let orgMap = new Map()
+let webpagestowebpagesId = new Map();
+let languageIdCodeMap = new Map();
+let websiteIdtoLanguage = new Map();
+
+
+export async function languageIdtoCodeMap(accessToken: string, dataverseOrg: any, entity: string) {
+
+    try {
+        const requestUrl = getCustomRequestURL(dataverseOrg, PORTAL_LANGUAGES, PORTAL_LANGUAGES_URL_KEY);
+        const req = await fetch(requestUrl, {
+            headers: getHeader(accessToken),
+        });
+
+        if (!req.ok) {
+            const options = { detail: 'Auth failed in language id code', modal: true };
+            vscode.window.showErrorMessage("Language code fetch failed", options);
+        }
+        const res = await req.json();
+        if (res) {
+            if (res.value.length > 0) {
+                for (let counter = 0; counter < res.value.length; counter++) {
+                    const adx_portallanguageid = res.value[counter].adx_portallanguageid ? res.value[counter].adx_portallanguageid : PORTAL_LANGUAGE_DEFAULT;
+                    const value = res.value[counter].adx_portallanguageid;
+                    languageIdCodeMap.set(adx_portallanguageid, value);
+                }
+            }
+        }
+    } catch (e: any) {
+        if (e.message === 'Unauthorized') {
+            const options = { detail: 'Auth failed in language id code', modal: true };
+            vscode.window.showErrorMessage("Language code fetch failed", options);
```

Comment:
this implementation is being repeated across every API call. Please move it a common function.